### PR TITLE
Add EIP-5170: Authenticity Proofs for Token Client Scripts

### PR DESCRIPTION
**Pull request by JamesSmartCell**
Provide a method to assert and revoke the authenticity of an ERC5169 client script for token contracts.

### CODE REVIEW

File: EIPS/eip-5170.md
Comment (Pandapip1) [inline EIPS/eip-5170.md:11]: Since this is very much linked to that EIP, I would recommend you merge this EIP with that one. Technically, this EIP cannot be merged until all EIPs in the `requires` list reach final.

Comment (MicahZoltu) [reply EIPS/eip-5170.md:11]: My general feeling is that if an EIP *can* be split into multiple EIPs it *should* be split into multiple EIPs.  If this is an extension that could be split out, I think that is the correct way forward.

> Technically, this EIP cannot be merged until all EIPs in the requires list reach final.

The rule is that an EIP can't pass the status of any EIP it depends on.  So if 5169 is in review, this can be in Draft or Review but not Last Call of Final.  This allows someone to walk multiple EIPs through the process at the same time when they have a dependency association between them.

**Review by JESSENIA1983 — APPROVED** [review]

File: EIPS/eip-5170.md
Comment (Pandapip1) [inline EIPS/eip-5170.md:28]:
```suggestion
[EIP-5169](./eip-5169.md) returns an all-purpose, one-version of script for use on the client side.
```
```suggestion
ERC 5169 returns an all-purpose, one-version of script for use on the client side.
```

File: EIPS/eip-5170.md
Comment (Pandapip1) [inline EIPS/eip-5170.md:22]:
```suggestion
Although [EIP-5169](./eip-5169.md) specified a way to obtain a set of client scripts through URI, it is inapplicable for token contracts that were issued before the creation of [EIP-5169](./eip-5169.md). Furthermore, it lacks the finesse to address situations such as
```
```suggestion
Although ERC 5169 specified a way to obtain a set of client scripts through URI, it is inapplicable for token contracts that was issued before the creation of ERC 5169. Furthermore, it lack the finess to address situations such as
```

File: EIPS/eip-5170.md
Comment (Pandapip1) [inline EIPS/eip-5170.md:4]:
```suggestion
description: Provide a method to assert and revoke the authenticity of an EIP-5169 client script for token contracts
```
```suggestion
description: Provide a method to assert and revoke the authenticity of an ERC5169 client script for token contracts
```

File: EIPS/eip-5170.md
Comment (Pandapip1) [inline EIPS/eip-5170.md:16]:
```suggestion
This EIP describes how to assert the authenticity of the script related to some token or smart contract, regardless of how the script was obtained.
```
```suggestion
This ERC describes how to assert the authenticity of the script related to some token or smart contract, regardless of how the script was obtained.
```

File: EIPS/eip-5170.md
Comment (Pandapip1) [inline EIPS/eip-5170.md:20]:
```suggestion
Often NFT authors want to provide some user functionality to their tokens through client scripts. This should be done safely, without opening the user to potential scams. Refer to [EIP-5169](./eip-5169.md) examples of such scripts.
```
```suggestion
Often NFT authors want to provide some user functionality to their tokens through client scripts. This should be done safely, without opening the user to potential scams. Refer to ERC 5169 examples of such scripts.
```

File: EIPS/eip-5170.md
Comment (Pandapip1) [inline EIPS/eip-5170.md:26]:
```suggestion
- A smart contract might have different scripts for different environments or use-cases. Take a subway token as an example. It might invoke a minimal script at the POS to be sent through NFC (Internet might be slow or inaccessible underground), while advanced functions for user retention, such as rewarding user mascot NFT for continued use, or carbon credit for buying carbon-neutral airfare.
- In a specific use case, a token's script is often localized, and it doesn't make sense to download and load all language translations.
- A specific use case might be compatible with only a specific version of the token's script.
```
```suggestion
- a smart contract might have different scripts for different environments or use-cases. Take a subway token as an example. It might invoke a minimal script at the POS to be sent through NFC (Internet might be slow or inaccessible underground), while advanced functions for user retention, such as rewarding user mascot NFT for continued use, or carbon credit for buying carbon-neutral airfare.
- in a specific use case, a token's script is often localized, and it doesn't make sense to download and load all language translations.
- a specific use-case might be compatible with only a specific version of the token's script.
```

File: EIPS/eip-5170.md
Comment (Pandapip1) [inline EIPS/eip-5170.md:34]:
```suggestion
Although the *token/smart contract author* and the *client script author* can be the same person/team, we will assume they are different people in this EIP, and the case that they are the same person/team can be implied.
```
```suggestion
Although the *token/smart contract author* and the *client script author* can be the same person/team, we will assume they are different people in this ERC, and the case that they are the same person/team can be implied.
```

File: EIPS/eip-5170.md
Comment (Pandapip1) [inline EIPS/eip-5170.md:52]:
```suggestion
1. Simplest (assuming internet connection and written *after* publication of this EIP):
```
```suggestion
1. Simplest (assuming internet connection and written *after* publication of this ERC):
```

File: EIPS/eip-5170.md
Comment (Pandapip1) [inline EIPS/eip-5170.md:107]:
```suggestion
This EIP does not specify how a wallet client obtains the signature as this can be realized in multiple ways. The simplest of which is to embed the client script in the `payload` of the JWS, as discussed above. This ensure that only a single URI is needed in order to locate both the client script, its signature and the signing key certificate. 
```
```suggestion
This ERC does not specify how a wallet client obtains the signature as this can be realized in multiple ways. The simplest of which is to embed the client script in the `payload` of the JWS, as discussed above. This ensure that only a single URI is needed in order to locate both the client script, its signature and the signing key certificate. 
```

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

**Review by Pandapip1 — CHANGES_REQUESTED** [review]
Please fix EIPW errors

**Comment by JamesSmartCell** [comment]
> Please fix EIPW errors

On hold until EIP-5169 is approved/merged, then this EIP can link to it. Without EIP-5169 this EIP is meaningless, so it makes sense to hold.

File: EIPS/eip-5170.md
Comment (Pandapip1) [inline EIPS/eip-5170.md:3]:
```suggestion
title: Authenticity Proofs for Token Client Scripts
```
Required: The title is too long. The title I have here is exactly 44 characters (the maximum)

File: EIPS/eip-5170.md
Comment (Pandapip1) [inline EIPS/eip-5170.md:6]: Required: Make a link on Ethereum Magicians

File: EIPS/eip-5170.md
Comment (Pandapip1) [inline EIPS/eip-5170.md:46]:
```suggestion
The authenticity of the client script may be obtained through the `scriptURI()` function call, as in EIP-5169, or supplied separately by the use-cases. However, this ERC is applicable to any code or data that is signed, and a client must validate the signature in the way specified in this ERC. In real-life use-cases, the client scripts can be either supplied by the aforementioned `scriptURI()` or offered to the client (wallet) in any way the wallet can work, even through NFC connections or QR codes.
```

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

**Comment by JamesSmartCell** [comment]
@Pandapip1 I don't know how to fix the Walidator error - the sections it's asking for exist.

**Comment by Pandapip1** [comment]
You're using `###` instead of `##`. Move all the sections up one header amount.

File: EIPS/eip-5170.md
Comment (xinbenlv) [inline EIPS/eip-5170.md:75]: ERCs usually specify an interface or behaviors in the specifications and then provide some reference implementation. Can you update the specification and implementation sections accordingly?

**Pull request closed without merging**
